templates/types/streaming/fastapi/app/engine/generate.py

```diff
@@ -54,14 +57,23 @@ def generate_datasource():
     ingestion_pipeline.vector_store = vector_store
 
     # Run the ingestion pipeline and store the results
-    ingestion_pipeline.run(show_progress=True, documents=documents)
+    nodes = ingestion_pipeline.run(show_progress=True, documents=documents)
 
     # Default vector store only keeps data in memory, so we need to persist it
     # Can remove if using a different vector store
     if isinstance(vector_store, SimpleVectorStore):
-        vector_store.persist(os.path.join(STORAGE_DIR, "vector_store.json"))
-    # Persist the docstore to apply ingestion strategy
-    docstore.persist(os.path.join(STORAGE_DIR, "docstore.json"))
+        index = VectorStoreIndex(
+            nodes=nodes,
+            storage_context=StorageContext.from_defaults(
+                docstore=docstore,
+                vector_store=vector_store,
+            ),
+            store_nodes_override=True,
+        )
+        index.storage_context.persist(STORAGE_DIR)
+    else:
+        # Persist the docstore to apply ingestion strategy
+        docstore.persist(os.path.join(STORAGE_DIR, "docstore.json"))
```

Comment:
i guess we only need to do this if we have a file-based docstore (which we're having at the moment anyhow only):
```suggestion
        # SimpleDocumentStore keeps the data in memory only - needs to be persisted explicitly 
        if isinstance(docstore, SimpleDocumentStore):
          docstore.persist(os.path.join(STORAGE_DIR, "docstore.json"))
```

Reply:
For now, we're using ingestion pipeline with a strategy which is required a docstore, and the file base docstore is default option. We also need it for all the vector store too.